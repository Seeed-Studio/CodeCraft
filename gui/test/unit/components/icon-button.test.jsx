import React from 'react';
import {shallow} from 'enzyme';
import IconButton from '../../../src/components/icon-button/icon-button';
import renderer from 'react-test-renderer';

describe('IconButtonComponent', () => {
    test('matches snapshot', () => {
        const onClick = jest.fn();
        const title = <div>Text</div>;
        const imgSrc = 'imgSrc';
        const className = 'custom-class-name';
        const component = renderer.create(
            <IconButton
                className={className}
                img={imgSrc}
                title={title}
                onClick={onClick}
            />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('triggers callback when clicked', () => {
        const onClick = jest.fn();
        const title = <div>Text</div>;
        const imgSrc = 'imgSrc';
        const componentShallowWrapper = shallow(
            <IconButton
                img={imgSrc}
                title={title}
                onClick={onClick}
            />
        );
        componentShallowWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });
});
