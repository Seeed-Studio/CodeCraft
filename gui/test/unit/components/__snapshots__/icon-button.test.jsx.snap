// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonComponent matches snapshot 1`] = `
<div
  className="custom-class-name"
  onClick={[Function]}
  role="button"
>
  <img
    className={undefined}
    draggable={false}
    src="imgSrc"
  />
  <div
    className={undefined}
  >
    <div>
      Text
    </div>
  </div>
</div>
`;
